HIP/hip-514-missing-hts-precompile-functions.md

```diff
@@ -0,0 +1,244 @@
+| `927da105`  | `allowance(address, address, address)`            | `(int256, uint256)`               |
+| `e1f21c67`  | `approve(address, address, uint256)`              | `(int256)`                        |
+| `7336aaf0`  | `approveNFT(address, address, uint256)`           | `(int256)`                        |
+| `098f2366`  | `getApproved(address, uint256)`                   | `(int256, address)`               |
+| `f49f40db`  | `isApprovedForAll(address, address, address)`     | `(int256, bool)`                  |
+| `367605ca`  | `setApprovalForAll(address, address, bool)`       | `(int256)`                        |
+| `46de0fb1`  | `isFrozen(address, address)`                      | `(int256, bool)`                  |
+| `f2c31ff4`  | `isKyc(address, address)`                         | `(int256, bool)`                  |
+| `f069f712`  | `deleteToken(address)`                            | `(int256)`                        |
+| `ae7611a0`  | `getTokenCustomFees(address)`                     | `(int256, CustomFee[])`           |
+| `a7daa18d`  | `getTokenDefaultFreezeStatus(address)`            | `(int256, bool)`                  |
+| `335e04c1`  | `getTokenDefaultKycStatus(address)`               | `(int256, bool)`                  |
+| `d614cdb8`  | `getTokenExpiryInfo(address)`                     | `(int256, Expiry)`                |
+| `3f28a19b`  | `getFungibleTokenInfo(address)`                   | `(int256, FungibleTokenInfo)`     |
+| `1f69565f`  | `getTokenInfo(address)`                           | `(int256, TokenInfo)`             |
+| `3c4dd32e`  | `getTokenKey(address, uint)`                      | `(int256, KeyValue)`              |
+| `2c20dcd1`  | `getNonFungibleTokenInfo(address, uint32)`        | `(int256, NonFungibleTokenInfo)`  |
+| `5b8f8584`  | `freezeToken(address, address)`                   | `(int256)`                        |
+| `a9b86b17`  | `unFreezeToken(address, address)`                 | `(int256)`                        |
+| `8f8d7f99`  | `grantTokenKyc(address, address)`                 | `(int256)`                        |
+| `af99c633`  | `revokeTokenKyc(address, address)`                | `(int256)`                        |
+| `7c41ad2c`  | `pauseToken(address)`                             | `(int256)`                        |
+| `3f8cc5c2`  | `unPauseToken(address)`                           | `(int256)`                        |
+| `9790686d`  | `wipeTokenAccount(address, address, uint32)`      | `(int256)`                        |
+| `1da62b01`  | `wipeTokenAccount(address, address, uint64[]])`   | `(int256)`                        |
+| `2cccc36f`  | `updateTokenInfo(address, HederaToken)`           | `(int256)`                        |
+| `593d6e82`  | `updateTokenExpiryInfo(address, Expiry)`          | `(int256)`                        |
+| `6fc3cbaf`  | `updateTokenKeys(address, uint, KeyValue)`        | `(int256)`                        |
```

Comment:
Currently we use `int256` for the return response code for the current token precompile methods. Probably now it's a good moment to switch to a more restricted value like `int16`? I believe the return response code values won't get too large and we can cover them with a more memory performant numeric type. Is there a specific reason we support `int` (`int256`) as return type?

Reply:
Aren't we already locked into the use of `int` since previous functions use it.
I know return types don't contribute to the hash and uniqueness of method selectors but is it safe practice to change return types.
What impact would it have on services consumption of the precompile contract?
@shemnon thoughts?

Reply:
It only matters if we pack responses and have multiple returns, or if we have numbers out of range.  Currently I don't think there is a way to have solidity pack return results.  So we could migrate to int64.

Reply:
Updated